components/console-backend-service/internal/domain/k8s/namespace_resolver_test.go

```diff
@@ -0,0 +1,147 @@
+package k8s_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/domain/k8s"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/domain/k8s/automock"
+	appAutomock "github.com/kyma-project/kyma/components/console-backend-service/internal/domain/shared/automock"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlerror"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestNamespaceResolver_NamespacesQuery(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		name := "name"
+		labels := map[string]string{
+			"env": "true",
+		}
+		resource := fixNamespace(name, labels)
+		resources := []*v1.Namespace{resource, resource}
+		expectedResult := gqlschema.Namespace{
+			Name:   name,
+			Labels: labels,
+		}
+
+		expected := []gqlschema.Namespace{
+			expectedResult, expectedResult,
+		}
+		svc := automock.NewNamespaceSvc()
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		svc.On("List").Return(resources, nil).Once()
+		defer svc.AssertExpectations(t)
+
+		converter := automock.NewNamespaceConverter()
+		converter.On("ToGQLs", resources).Return(expected, nil).Once()
+		defer converter.AssertExpectations(t)
+
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
+		resolver.SetNamespaceConverter(converter)
+
+		result, err := resolver.NamespacesQuery(nil, nil)
+
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("Empty", func(t *testing.T) {
+		resources := []*v1.Namespace{}
+		expected := []gqlschema.Namespace(nil)
+		svc := automock.NewNamespaceSvc()
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		svc.On("List").Return(resources, nil).Once()
+		defer svc.AssertExpectations(t)
+
+		converter := automock.NewNamespaceConverter()
+		converter.On("ToGQLs", resources).Return(expected, nil).Once()
+		defer converter.AssertExpectations(t)
+
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
+		resolver.SetNamespaceConverter(converter)
+
+		result, err := resolver.NamespacesQuery(nil, nil)
+
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("Error", func(t *testing.T) {
```

Comment:
If test below is named `ErrorConverting`, maybe this could be named `ErrorListing`?